msmbuilder/featurizer/featurizer.py

```diff
@@ -307,6 +307,75 @@ def partial_transform(self, traj):
         return d ** self.exponent
 
 
+class FunctionFeaturizer(Featurizer):
+    """Featurizer based on arbitrary functions.
+
+    This featurizer transforms a dataset containing MD trajectories into
+    a vector dataset by representing each frame in each of the MD trajectories
+    by a vector the output of the function.
+
+    Parameters
+    ----------
+    function : function
+        Instantiation of the function. The function should accept
+        a mdtraj.Trajectory object as the first argument.
+    func_args : dictionary
+        A dictionary of key word arguments(keys) and their values to
+        pass to the function. These should NOT include the trajectory
+        object which is passed in as the first argument.
+
+    Notes
+    ----------
+    This Featurizer assumes that the function takes in the trajectory object
+    as the first argument.
+
+    Examples
+    --------
+    >>> function = compute_dihedrals
+    >>> f = FunctionFeaturizer(function, func_args={indices: [[0,1,2,3]]})
+    >>> results = f.transform(dataset)
+    """
+
+    def __init__(self, function, func_args={}):
+        if callable(function):
+            self.function = function
+            self.func_args = func_args
+        else:
+            raise Exception("Sorry but we "
```

Comment:
This should probably be `ValueError` and maybe include the fact that the object isn't callable in the error message so the user knows what's wrong